libbpfgo.go

```diff
@@ -1711,6 +1711,35 @@ func doAttachUprobe(prog *BPFProg, isUretprobe bool, pid int, path string, offse
 	return bpfLink, nil
 }
 
+type AttachFlag uint32
+
+const (
+	BPFFNone          AttachFlag = 0
+	BPFFAllowOverride AttachFlag = C.BPF_F_ALLOW_OVERRIDE
+	BPFFAllowMulti    AttachFlag = C.BPF_F_ALLOW_MULTI
+	BPFFReplace       AttachFlag = C.BPF_F_REPLACE
+)
+
+// Attach attaches the BPFProgram to a targetFd at the specified attachType hook.
+func (p *BPFProg) Attach(targetFd int, attachType BPFAttachType, flags AttachFlag) error {
```

Comment:
We have `AttachGeneric` which is similar, but does not allow to specify a fd to attach to. Should we name this `AttachGenericFD`?

Reply:
I also see that you're using it for attaching a program to a cgroupFD (Which could be done by the attach_cgroup method, no ?). I want to understand if:

- should we make attachgeneric generic enough to allow a fd to attach to ? (or is it just the cgroup case and then we already have an attachment function for it that could be used and/or changed)

- this is a specific need for something else (then we should either change AttachGeneric OR the selftest to differentiate between the cgroup attachment).


WDYT ?

Reply:
> We have `AttachGeneric` which is similar, but does not allow to specify a fd to attach to. Should we name this `AttachGenericFD`?

SGTM. I'll rename it to `AttachGenericFD`.

> I also see that you're using it for attaching a program to a cgroupFD (Which could be done by the attach_cgroup method, no ?)

That program can't be attached by the attach_cgroup method.

> should we make attachgeneric generic enough to allow a fd to attach to ? (or is it just the cgroup case and then we already have an attachment function for it that could be used and/or changed)

That would be a breaking change and would make the attachgeneric method very complex. we should avoid it.

> this is a specific need for something else (then we should either change AttachGeneric OR the selftest to differentiate between the cgroup attachment).

Yes, This is a specific method for attaching cgroup, sockops, flow_dissector, lirc_mode2, and sockmap programs. 
I'll change the selftest to attach a sockmap program.